vault/logical_system_test.go

```diff
@@ -5113,3 +5113,35 @@ func TestSortVersionedPlugins(t *testing.T) {
 		})
 	}
 }
+
+func TestValidateVersion(t *testing.T) {
+	b := testSystemBackend(t).(*SystemBackend)
+	k8sAuthBuiltin := versions.GetBuiltinVersion(consts.PluginTypeCredential, "kubernetes")
+
+	for name, tc := range map[string]struct {
+		pluginName         string
+		pluginVersion      string
+		pluginType         consts.PluginType
+		expectLogicalError bool
+		expectedVersion    string
+	}{
+		"default, nothing in nothing out": {"kubernetes", "", consts.PluginTypeCredential, false, ""},
+		"builtin specified, empty out":    {"kubernetes", k8sAuthBuiltin, consts.PluginTypeCredential, false, ""},
+		"not canonical is ok":             {"kubernetes", "1.0.0", consts.PluginTypeCredential, false, "v1.0.0"},
+		"not a semantic version, error":   {"kubernetes", "not-a-version", consts.PluginTypeCredential, true, ""},
```

Comment:
It doesn't look like we are testing for the shadowed built-in case?

Reply:
Yep, sorry I should have mentioned the tests are still WIP. I'll be adding more today.